rust/examples/otel-stdout-components.rs

```diff
@@ -0,0 +1,92 @@
+use dylibso_observe_sdk::{adapter::otelstdout::OtelStdoutAdapter, context::component::ObserveSdkBindings};
+
+use wasmtime::component::Val;
+use wasmtime_wasi::preview2::{WasiView, WasiCtx, Table};
+
+struct State {
+    table: Table,
+    wasi_ctx: WasiCtx,
+    observe_sdk_bindings: ObserveSdkBindings,
+}
+
+impl WasiView for State {
+    fn table(&self) -> &Table {
+        &self.table
+    }
+
+    fn table_mut(&mut self) -> &mut Table {
+        &mut self.table
+    }
+
+    fn ctx(&self) -> &WasiCtx {
+        &self.wasi_ctx
+    }
+
+    fn ctx_mut(&mut self) -> &mut WasiCtx {
+        &mut self.wasi_ctx
+    }
+}
+
+impl AsMut<ObserveSdkBindings> for State {
+    fn as_mut(&mut self) -> &mut ObserveSdkBindings {
+        &mut self.observe_sdk_bindings
+    }
+}
+
+#[tokio::main]
+pub async fn main() -> anyhow::Result<()> {
+    let args: Vec<_> = std::env::args().skip(1).collect();
+    let data = std::fs::read(&args[0])?;
+    let function_name = "hello-world";
+    let mut config = wasmtime::Config::new();
+
+    config.async_support(true);
+    config.wasm_component_model(true);
+
+    // Create instance
+    let engine = wasmtime::Engine::new(&config)?;
+    let component = wasmtime::component::Component::new(&engine, &data)?;
+
+    let mut table = Table::new();
+
+    // Setup WASI
+    let wasi_ctx = wasmtime_wasi::preview2::WasiCtxBuilder::new()
+        .inherit_stdio()
+        .args(&args.clone())
+        .build(&mut table)?;
+
+    let mut linker = wasmtime::component::Linker::new(&engine);
+
+    let adapter = OtelStdoutAdapter::create();
+    // Provide the observability functions to the `Linker` to be made available
+    // to the instrumented guest code. These are safe to add and are a no-op
+    // if guest code is uninstrumented.
+    let (observe_sdk_bindings, trace_ctx) = adapter.create_bindings(&data, Default::default())?;
```

Comment:
I had to split "create the bindings" from "start"; we have to store the bindings in state. (I played with `Box::leak`'ing them – or otherwise squirreling them away elsewhere in memory – but ran headlong into an argument with rustc. I lost the argument.)